test/com/amazon/ion/system/IonReaderBuilderTest.java

```diff
@@ -187,4 +191,40 @@ public void testIncrementalReadingSupportsAutoGzip() throws IOException
         reader2.close();
     }
 
+    @Test
+    public void concatenatedAfterGZIPHeader() throws Exception {
```

Comment:
This makes 8 uses of the bytes-to-GZIP idiom, definitely calling for a helper method. Same feedback on ordering of input construction and expectations.

Reply:
Accepted your proposal below.